packages/marshal/src/helpers/promise.js

```diff
@@ -22,27 +27,40 @@ export const PromiseHelper = harden({
   styleName: 'promise',
 
   canBeValid: (candidate, check) => {
+    const passStyleDesc = getOwnPropertyDescriptor(candidate, PASS_STYLE);
     return (
-      (candidate[PASS_STYLE] === 'promise' ||
+      ((passStyleDesc && passStyleDesc.value === 'promise') ||
       isPromise(candidate) ||
-      checkTagRecord(candidate, 'promise', check))
+      check(
+        false,
+        X`Pseudo-promise must be an object with ${q(PASS_STYLE)} ${q(
+          'promise',
+        )}: ${candidate}`,
+      ))
     );
   },
 
   assertValid: candidate => {
-    PromiseHelper.canBeValid(candidate, assertChecker);
     if (isPromise(candidate)) {
       assertSafePromise(candidate);
       return;
     }
 
-    const proto = getPrototypeOf(candidate);
-    proto === objectPrototype ||
-      proto === null ||
+    checkTagRecord(candidate, 'promise', assertChecker);
+
+    // XXX Should this (and TaggedHelper.assertValid) support a null prototype?
+    getPrototypeOf(candidate) === objectPrototype ||
       assert.fail(X`Unexpected prototype for: ${candidate}`);
```

Comment:
Good question. Yes, I think both should. But here's a counter-argument that says we shouldn't support it at all, i.e., that we should remove such support even from copyRecords:

Any such non-standard `null` prototype is not pass invariant. If such a copyRecord is passed through any marshal barrier, the copyRecord that gets unserialized on the other side will always inherit directly from `Object.prototype'.

So, I am still on the fence. Opinions?

Reply:
I am in favor of removing support for all null-prototype passables unless we have a clear scenario benefiting from them, in which case every check for Object.prototype should be expanded accordingly.

Reply:
I agree. So no change suggested here. Anytime you'd like to propagate this reform (assuming it doesn't break anything but tests), I'll happily review. Thanks!

Reply:
#1324

Reply:
Necroing this thread but an argument could be made that a copyRecord should be unmarshalled with a null prototype instead.

Reply:
Given that we agree on preserving pass-invariance, there are only two solutions:
   1. All CopyRecords inherit from `Object.prototype` (status quo)
   2. All CopyRecords inherit from `null` (what we’d all prefer if we could have it everywhere at reasonable cost)

Yes, we could change `passStyleOf` and `unserialize` to be consistent with (2). What we cannot reasonably change is requiring everyone to write
```js
harden({
  __proto__: null,
  foo: 3
})
```
rather than the status quo
```js
harden({foo: 3})
```
Yes, there are approaches to this notational constraint, like using something instead of `harden` that either
   * modifies the argument record in place (like `Far` does), or
   * produces a new object derived from the argument object
Altogether, for the way we use CopyRecords everywhere, I don’t think either of these coping strategies are at reasonable notational and explanatory cost. Such a cure would be worse than the disease.